sound/soc/codecs/hdac_hda.h

```diff
@@ -6,14 +6,22 @@
 #ifndef __HDAC_HDA_H__
 #define __HDAC_HDA_H__
 
+#define HDAC_ANALOG_DAI_ID		0
+#define HDAC_DIGITAL_DAI_ID		1
+#define HDAC_ALT_ANALOG_DAI_ID		2
+#define HDAC_HDMI_0_DAI_ID		3
+#define HDAC_HDMI_1_DAI_ID		4
+#define HDAC_HDMI_2_DAI_ID		5
+#define HDAC_LAST_DAI_ID		6
+
 struct hdac_hda_pcm {
 	int stream_tag[2];
 	unsigned int format_val[2];
 };
 
 struct hdac_hda_priv {
 	struct hda_codec codec;
-	struct hdac_hda_pcm pcm[2];
+	struct hdac_hda_pcm pcm[HDAC_LAST_DAI_ID];
```

Comment:
This is another hiding bug. Already before this patch there was AltAnalog id defined, but the above array was only sized for two. However one could not hit the issue before.